cmd/podman/ps.go

```diff
@@ -506,41 +551,39 @@ func getTemplateOutput(containers []*libpod.Container, opts batchcontainer.PsOpt
 		}
 		psOutput = append(psOutput, params)
 	}
-	// Sort the ps entries by created time
-	sort.Sort(psSorted(psOutput))
 
 	return psOutput, nil
 }
 
-// getJSONOutput returns the container info in its raw form
-func getJSONOutput(containers []*libpod.Container, opts batchcontainer.PsOptions) ([]psJSONParams, error) {
+// getAndSortJSONOutput returns the container info in its raw, sorted form
+func getAndSortJSONParams(containers []*libpod.Container, opts batchcontainer.PsOptions) ([]psJSONParams, error) {
 	var (
-		psOutput []psJSONParams
+		psOutput psSorted
 		ns       *batchcontainer.Namespace
 	)
 	for _, ctr := range containers {
 		batchInfo, err := batchcontainer.BatchContainerOp(ctr, opts)
 		if err != nil {
 			return nil, err
 		}
+
 		if opts.Namespace {
 			ns = batchcontainer.GetNamespaces(batchInfo.Pid)
 		}
 		params := psJSONParams{
-			ID:               ctr.ID(),
-			Image:            batchInfo.ConConfig.RootfsImageName,
-			ImageID:          batchInfo.ConConfig.RootfsImageID,
-			Command:          batchInfo.ConConfig.Spec.Process.Args,
-			CreatedAt:        batchInfo.ConConfig.CreatedTime,
-			Status:           batchInfo.ConState.String(),
-			Ports:            batchInfo.ConConfig.PortMappings,
-			RootFsSize:       batchInfo.RootFsSize,
-			RWSize:           batchInfo.RwSize,
-			Names:            batchInfo.ConConfig.Name,
-			Labels:           batchInfo.ConConfig.Labels,
-			Mounts:           batchInfo.ConConfig.Spec.Mounts,
-			ContainerRunning: batchInfo.ConState == libpod.ContainerStateRunning,
```

Comment:
Why is this dropped?

Reply:
It's been added back to this function because previously I was formatting the JSON based on an intermediary struct psParams, that did the whole batchInfo conversion. Now, I do this conversion directly to the JSON, and format the template parameters off of the JSON. One less step of redirection if it is json.